### PR TITLE
Add optional fields school and acad streams

### LINKED ISSUE #27: As a tutor, I can to track my students' schools so that I can teach in accordance with their syllabus.

### PR DESCRIPTION
**Pull request by Xiaoyunnn**
Fix #27

### CODE REVIEW

**Review by angkoonhwee — COMMENTED** [review]
Looks good to me atm! Just that some parts feel quite repetitive but I also don't have a good idea of how to change it so I think we can just ignore them first >,<

File: src/main/java/seedu/address/model/AddressBook.java [hunk 1]
```diff
@@ -97,7 +97,8 @@ public void removePerson(Person key) {
 
     @Override
     public String toString() {
-        return persons.asUnmodifiableObservableList().size() + " persons";
+        return persons.asUnmodifiableObservableList().size() + " persons\n"
+                + persons;
         // TODO: refine later
```
Comment (eeliana) [inline src/main/java/seedu/address/model/AddressBook.java:102]: What is this for?

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 2]
```diff
@@ -151,11 +159,29 @@ public Person toModelType() throws IllegalValueException {
         }
         final Address modelAddress = new Address(address);
 
+        if (school == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, School.class.getSimpleName()));
+        }
+        if (!school.isEmpty() && !School.isValidSchName(school)) {
```
Comment (lingshanng) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:165]: check `isEmpty` inside `isValidSchName`
```suggestion
        if (!School.isValidSchName(school)) {
```

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 3]
```diff
@@ -151,11 +159,29 @@ public Person toModelType() throws IllegalValueException {
...
+            throw new IllegalValueException(School.MESSAGE_CONSTRAINTS);
+        }
+        final School modelSchool = new School(school);
+
+        if (acadStream == null) {
+            throw new IllegalValueException(
+                    String.format(MISSING_FIELD_MESSAGE_FORMAT, AcadStream.class.getSimpleName()));
+        }
+        if (!acadStream.isEmpty() && !AcadStream.isValidAcadStream(acadStream)) {
```
Comment (lingshanng) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:174]:
```suggestion
        if (!AcadStream.isValidAcadStream(acadStream)) {
```

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -95,11 +97,41 @@ public static Email parseEmail(String email) throws ParseException {
...
+    /**
+     * Parses a {@code String school} into a {@code School}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code school} is invalid.
+     */
+    public static School parseSchool(String school) throws ParseException {
+        requireNonNull(school);
+        String strippedSchName = school.strip();
+        if (!strippedSchName.isEmpty() && !School.isValidSchName(strippedSchName)) {
```
Comment (lingshanng) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:116]:
```suggestion
        if (!School.isValidSchName(strippedSchName)) {
```

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 5]
```diff
@@ -95,11 +97,41 @@ public static Email parseEmail(String email) throws ParseException {
...
+    /**
+     * Parses a {@code String acadStream} into an {@code AcadStream}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code acadStream} is invalid.
+     */
+    public static AcadStream parseAcadStream(String acadStream) throws ParseException {
+        requireNonNull(acadStream);
+        String strippedAcadStream = acadStream.strip();
+        if (!strippedAcadStream.isEmpty() && !AcadStream.isValidAcadStream(strippedAcadStream)) {
```
Comment (lingshanng) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:131]:
```suggestion
        if (!AcadStream.isValidAcadStream(strippedAcadStream)) {
```

File: src/main/java/seedu/address/model/person/AcadStream.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+public class AcadStream {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Academic stream should only contain alphabetic characters and spaces.";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     * Academic stream can be any alphabetic character with or without space.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alpha}][\\p{Alpha} ]*";
```
[the remark this answers is not in the document]
Comment (Xiaoyunnn) [reply src/main/java/seedu/address/model/person/AcadStream.java:19]: why tho? in what cases do we need other characters?

**Comment by Xiaoyunnn** [comment]
why need double quotations? @Chesterwongz

**Review by Chesterwongz — COMMENTED** [review]
sorry, it just felt weird to me to allow 'O' Level but not "O" Level, I feel this is the problem with limiting what is allowed, you need to think of all the possibilities, or justify why we aren't allowing certain things... And if you want to allow as many possibilities as possible why not just have no limitations? @Xiaoyunnn

**Review by lingshanng — APPROVED** [review]
LGTM

**Pull request merged**